lib/openxml/drawingml/elements/paragraph.rb

```diff
@@ -0,0 +1,74 @@
+module OpenXml
+  module DrawingML
+    module Elements
+      class Paragraph < OpenXml::DrawingML::Element
+        include HasChildren
+        include HasProperties
+
+        tag :p
+
+        # Validations
+        COORDINATE_REGEX = /^(-?[0-9]+(:?\.[0-9]+)?)(:?mm|cm|in|pt|pc|pi)?$/
+
+        coordinate = ->(value) { raise ArgumentError unless value.to_s =~ coordinate_regex }
+        indent_range = lambda { |value|
+          matched = value.to_s =~ COORDINATE_REGEX
+          raise ArgumentError unless matched || (-51_206_400..51_206_400).cover?(value)
+        }
+        text_margin = lambda { |value|
+          matched = value.to_s =~ COORDINATE_REGEX
+          raise ArgumentError unless matched || (0..51_206_400).cover?(value)
+        }
+
+
+        properties_attribute :alignment, displays_as: :algn, one_of: %i{ ctr dist just justLow l r thaiDist }
+        properties_attribute :default_tab_size, displays_as: :defTabSz, validation: coordinate
+        properties_attribute :east_asian_line_break, displays_as: :eaLnBrk, expects: :boolean
+        properties_attribute :vertical_alignment, displays_as: :fontAlgn, one_of: %i{ auto b base ctr t }
+        properties_attribute :hanging_punctuation, displays_as: :hangingPunct, expects: :boolean
+        properties_attribute :indent, validation: indent_range
+        properties_attribute :latin_line_break, displays_as: :latinLnBrk, expects: :boolean
+        properties_attribute :level, displays_as: :lvl, in_range: 0..8
+        properties_attribute :left_margin, displays_as: :marL, validation: text_margin
+        properties_attribute :right_margin, displays_as: :marR, validation: text_margin
+        properties_attribute :right_to_left, displays_as: :rtl, expects: :boolean
+
+        property :line_spacing
+        property :space_before
+        property :space_after
+
+        property_choice do
+          property :bullet_color
+          value_property :bullet_color_matches_text, as: :bullet_color_text
+        end
+
+        property_choice do
+          value_property :bullet_size_matches_text, as: :bullet_size_text
+          value_property :bullet_size_percent
+          value_property :bullet_size_points
+        end
+
+        property_choice do
+          value_property :bullet_font_matches_text, as: :bullet_font_text
+          value_property :bullet_font
+        end
+
+        property_choice do
+          value_property :disable_bullets, as: :bullet_style_none
+          property :autonumbered_bullets, as: :bullet_style_auto_numbered
+          property :character_bullets, as: :bullet_style_character
+          property :blip_bullets, as: :bullet_style_blip
+        end
+
+        property :tab_list
+        property :default_run_properties
+        property :extension_list
+
+      private
```

Comment:
Is this `private` and the whitespace below it here for any reason?

Reply:
I think this was copy/pasted from the element I moved over from Pptx